.github/workflows/ci_clang_format_check.yml

```diff
@@ -7,16 +7,23 @@ jobs:
     strategy:
       matrix:
         path:
-          - 'api'
-          - 'bin'
-          - 'stuffer'
-          - 'error'
-          - 'tls/extensions'
-          - 'utils'
+          - check: 'api'
+            exclude: ''
+          - check: 'bin'
+            exclude: ''
+          - check: 'stuffer'
+            exclude: ''
+          - check: 'error'
+            exclude: ''
+          - check: 'tls/extensions'
```

Comment:
Might as well add the first round of tls changes to this PR. That would prove that this exclusion regex is working.

Reply:
Okay. I added the tls/ sub-folder with the exclusion regex to exclude `tls/s2n_*`.

I removed `tls/extensions` which now should be checked by the tls/ and should not be excluded.

Reply:
I also re-based on main which adds a entry here.

Reply:
Is this working? I see that the formatting check for the extensions folder didn't finish executing.

Reply:
Yes. The tls/extensions format checker didn't run because it's been removed in this PR. It's been subsumed by the tls checker which this PR adds.

Reply:
Okay looks like it is marked as required on the CI. I'll add tls/extensions back. It will double check, but that's okay,

Reply:
I think Madeleine made a good point-- can we include the first batch of tls changes (maybe just s2n_a*?) to prove this will work?

Reply:
Yup, ok. Starting with `tests/unit/s2n_[3a][a-e]*\.c`

Reply:
I also moved to this multi-line string format in yaml. The string itself hasn't changed. The newlines and backslashes are escaped and the leading white space is insignificant. But now future PRs won't conflict with each other.

Reply:
Discussed with Harrison. He is going to attempt to clean up this test/unit regex so that it excludes everything except the group that is currently being formatted (in one line instead of like 200). If that doesn't work we will break this folder into larger groups of like 50 files each and see if we can get those through PR with less pain. As is this regex is too complex to commit.

Reply:
Done.

Reply:
The new strategy is to just go letters at a time through the alphabet. This will result in including larger numbers of files per pr. But per our discussion --> that is okay.